docs/specification.md

```diff
@@ -833,6 +833,47 @@ nat
 
 # TZIP-16 metadata
 
-We do not provide a proper support for contract metadata until [the proposal](https://gitlab.com/tzip/tzip/-/merge_requests/94) is finalized and merged.
+Note: we do not provide a proper support for contract metadata until [the proposal](https://gitlab.com/tzip/tzip/-/merge_requests/94) is finalized and merged.
 
-Except for basic fields, we are going to include `GetCounter` off-chain view.
+This contract implements TZIP-16.
+
+The DAO contract itself won't store the metadata, rather a dedicated contract will contain that.
+Motivation:
+* The baseDAO contract is very large and approaches the Tezos hard limits.
+  * Providing the metadata during origination would significantly reduce the amount of functionality we can put into contract;
+  * Inserting the metadata after origination still requires mechanisms to manage the metadata;
+    this way there is no guarantee that it remains constant (as users will probably expect).
+
+## Deployment with metadata
```

Comment:
Shouldn't it be in the usage instructions? This specification is supposed to be more or less language/tool independent, but here we are referring to `stack` and implementation details.